backend/moonstreamapi/routes/queries.py

```diff
@@ -0,0 +1,384 @@
+    except Exception as e:
+        logger.error(
+            f"Error listing subscriptions for user ({request.user.id}) with token ({request.state.token}), error: {str(e)}"
+        )
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    used_queries: List[str] = [
+        resource.resource_data["name"] for resource in resources.resources
+    ]
+
+    query_name = slugify(query_applied.name)
+
+    if query_name in used_queries:
+
+        raise MoonstreamHTTPException(
+            status_code=404,
+            detail=f"Provided query name already use. Please remove it or use PUT /{query_name}",
+        )
+
+    try:
+        # Put query to journal
+        entry = bc.create_entry(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            title=f"Query:{query_name}",
+            tags=["type:query"],
+            content=query_applied.query,
+        )
+    except BugoutResponseException as e:
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(f"Error creating query entry: {str(e)}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    try:
+        # create resource query_name_resolver
+        bc.create_resource(
+            token=token,
+            application_id=MOONSTREAM_APPLICATION_ID,
+            resource_data={
+                "type": BUGOUT_RESOURCE_QUERY_RESOLVER,
+                "user_id": str(user.id),
+                "user": str(user.username),
+                "name": query_name,
+                "entry_id": str(entry.id),
+            },
+        )
+    except BugoutResponseException as e:
+        logger.error(f"Error creating name resolving resource: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(f"Error creating name resolving resource: {str(e)}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    try:
+
+        bc.update_tags(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            entry_id=entry.id,
+            tags=[f"query_id:{entry.id}", f"preapprove"],
+        )
+
+    except BugoutResponseException as e:
+        logger.error(f"Error in applind tags to query entry: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(f"Error in applind tags to query entry: {str(e)}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.get("/{query_name}/query", tags=["queries"])
+async def get_query_handler(request: Request, query_name: str) -> BugoutJournalEntry:
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)

```

Comment:
Better to wrap it to `try..catch..` block to handle resources fetch and data validation problems. For example dictinary could be formed incorrect and had no ["name"] key, they it will rase end error.